### PR TITLE
Rename SectionAuth to SectionSigned, the other SectionAuth to a simpler SectionSig

### PR DESCRIPTION
**Pull request by grumbach**
There are two different `SectionAuth` in `sn_interface`, both in the messaging module:
- one is a section sig with a Xorname: `SectionAuth`
- the other is a section sig along with the data it signed: `SectionAuth<T>`

This can be confusing because they are quite similar, have the same name and sometimes results in awkward aliasing situations when we have both in the scope:

```rust
use sn_interface::messaging::{
    system::SectionAuth,
    SectionAuth as MsgKindSectionAuth,
}
```

This PR renames the second one to `SectionSigned<T>` as I believe it makes more sense:

```rust
SectionSigned<SectionAuthorityProvider>
SectionSigned<NodeInfo>
...

```

It also merges the other `SectionAuth` with the `KeyedSig` into a single `SectionSig` type, thus removing the mostly unused `XorName` which can be faked anyway. This change helps simplify a lot of the code out. 

- `SectionAuth<T>` -> `SectionSigned<T>`
- `SectionAuth` -> `SectionSig`
- `KeyedSig` -> `SectionSig`
- `SigShare` -> `SectionSigShare`

### CODE REVIEW

File: sn_node/src/comm/listener.rs [hunk 1]
```diff
@@ -66,7 +66,14 @@ impl MsgListener {
                         }
                     };
 
-                    let src_name = wire_msg.auth().src_name();
+                    let src_name = match wire_msg.auth() {
+                        // NB TODO is default acceptable??
+                        AuthKind::SectionShare(_) => Default::default(),
```
Comment (grumbach) [inline sn_node/src/comm/listener.rs:71]: This is temporary

File: sn_client/src/api/register_apis.rs [hunk 2]
```diff
@@ -246,20 +246,16 @@ impl Client {
 }
 
 // temp dummy
-fn section_auth() -> sn_interface::messaging::SectionAuth {
-    use sn_interface::messaging::system::KeyedSig;
+fn section_auth() -> sn_interface::messaging::SectionSig {
```
Comment (davidrusu) [inline sn_client/src/api/register_apis.rs:249]: Rename the function too?
```suggestion
fn section_sig() -> sn_interface::messaging::SectionSig {
```

File: sn_interface/src/messaging/data/register.rs [hunk 3]
```diff
@@ -83,7 +83,7 @@ pub enum RegisterCmd {
         cmd: SignedRegisterCreate,
         /// Section signature over the operation,
         /// verifying that it was paid for.
-        section_auth: SectionAuth,
+        section_auth: SectionSig,
```
Comment (davidrusu) [inline sn_interface/src/messaging/data/register.rs:86]:
```suggestion
        section_sig: SectionSig,
```

File: sn_interface/src/messaging/network_knowledge.rs [hunk 4]
```diff
@@ -103,6 +103,6 @@ pub struct SectionsDAG {
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
 pub struct SectionTreeUpdate {
     pub section_auth: SectionAuthorityProvider,
-    pub section_signed: KeyedSig,
+    pub section_signed: SectionSig,
```
Comment (davidrusu) [inline sn_interface/src/messaging/network_knowledge.rs:106]: `section_signed` is confusing now that we have a `SectionSigned` struct, let's rename to `section_sig`
```suggestion
    pub section_sig: SectionSig,
```

Comment (grumbach) [reply sn_interface/src/messaging/network_knowledge.rs:106]: agreed!

File: sn_interface/src/network_knowledge/section_authority_provider.rs [hunk 5]
```diff
@@ -263,9 +263,9 @@ impl SectionAuthorityProviderMsg {
     }
 }
 
-impl SectionAuth<SectionAuthorityProviderMsg> {
-    pub fn into_authed_state(self) -> SectionAuth<SectionAuthorityProvider> {
-        SectionAuth {
+impl SectionSigned<SectionAuthorityProviderMsg> {
+    pub fn into_authed_state(self) -> SectionSigned<SectionAuthorityProvider> {
```
Comment (davidrusu) [inline sn_interface/src/network_knowledge/section_authority_provider.rs:267]:
```suggestion
    pub fn into_signed_state(self) -> SectionSigned<SectionAuthorityProvider> {
```

File: sn_node/src/node/messaging/agreement.rs [hunk 6]
```diff
@@ -127,8 +127,8 @@ impl Node {
     #[instrument(skip(self), level = "trace")]
     pub(crate) async fn handle_new_elders_agreement(
         &mut self,
-        signed_sap: SectionAuth<SectionAuthorityProvider>,
-        key_sig: KeyedSig,
+        signed_sap: SectionSigned<SectionAuthorityProvider>,
+        key_sig: SectionSig,
```
Comment (davidrusu) [inline sn_node/src/node/messaging/agreement.rs:131]:
```suggestion
        section_sig: SectionSig,
```

**Review by davidrusu — APPROVED** [review]
:ship:

**Pull request merged**